tests/args_parser.py

```diff
@@ -16,6 +16,9 @@ def parse_args(self):
         parser = argparse.ArgumentParser()
         parser.add_argument('--dev',
                             help='RDMA device to run the tests on')
+        parser.add_argument('--port',
+                            help='Use port <port> of RDMA device', type=int,
+                            default=1, choices=[1, 2])
```

Comment:
choices=[1, 2] <- chellsio supports 4 ports and Mellanox has such multiport device too.
It is better to be based on max supported by the device.

So I suggest to simply drop this "choices=[]" for now.

Reply:
This is needed to validate that the user is passing a valid port number, So instead of dropping the choices option, what about changing it to [1,2,3,4]?

Reply:
If user provided bad port, pyverbs test will crash.
Why isn't good enough?

Reply:
> If user provided bad port, pyverbs test will crash.
> Why isn't good enough?

This is a very bad user experience :-), running the whole pyverbs tests and later to understand the user passed a wrong port number...

Reply:
We are talking about pyverbs tests, even today with choices=[1,2], it will crash if i try to run over single port device.

IMHO, it just doesn't worth to make it "clean".

Reply:
I think Leon is right... I would just drop the choices.
As he mentioned, even now passing port num 2 on a single port device would fail the tests. The querying for the max ports option would complicate things and is not worth it right now.

Reply:
done.